runner/internal/metrics/aggregate.go

```diff
@@ -45,11 +57,47 @@ func Compute(records []recorder.Record) (agg Aggregate) {
 		}
 	}
 
-	agg.ResponseTimes = timestats.Compute(times)
+	agg.ResponseTimes = timestats.New(times)
 
-	agg.StatusCodeDistribution = computeStatusCodesDistribution(records)
+	agg.StatusCodesDistribution = computeStatusCodesDistribution(records)
 
 	agg.RequestEventTimes = computeRequestEventTimes(records)
 
 	return agg
 }
+
+// Special compute helpers.
+
+func computeRequestEventTimes(records []recorder.Record) map[string]timestats.TimeStats {
+	events := getFlatRelativeTimeEvents(records)
+
+	timesByEvent := map[string][]time.Duration{}
+
+	for _, e := range events {
+		timesByEvent[e.Name] = append(timesByEvent[e.Name], e.Time)
+	}
+
+	statsByEvent := map[string]timestats.TimeStats{}
+
+	for e, times := range timesByEvent {
+		statsByEvent[e] = timestats.New(times)
+	}
+
+	return statsByEvent
+}
+
+func getFlatRelativeTimeEvents(records []recorder.Record) []recorder.Event {
+	events := []recorder.Event{}
+	for _, record := range records {
+		events = append(events, record.RelativeTimeEvents()...)
+	}
+	return events
+}
+
+func computeStatusCodesDistribution(records []recorder.Record) map[string]int {
+	statuses := map[string]int{}
+	for _, rec := range records {
+		statuses[strconv.Itoa(rec.Code)]++
+	}
+	return statuses
+}
```

Comment:
Why does it have to be `map[string]int`? `http.Response.StatusCode` being `int`, `map[int]int` makes more sense to me, and also avoid a conversion here.
```suggestion
func computeStatusCodesDistribution(records []recorder.Record) map[int]int {
	statuses := map[int]int{}
	for _, rec := range records {
		statuses[rec.Code]++
	}
	return statuses
}
```